jbmc/src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -154,11 +155,17 @@ class java_bytecode_convert_classt:public messaget
   /// [overlay method](\ref java_bytecode_convert_classt::is_overlay_method)
   /// or an ignored method.
   ///
+  /// \param class_name: class the method is declared on
   /// \param method: a `methodt` object from a java bytecode parse tree
   /// \return true if the method is an ignored method, else false
-  static bool is_ignored_method(const methodt &method)
+  static bool is_ignored_method(
+    const irep_idt &class_name, const methodt &method)
   {
-    return method.has_annotation(ID_ignored_method);
+    static irep_idt org_cprover_CProver_name = "org.cprover.CProver";
+    return
+      (class_name == org_cprover_CProver_name &&
+       cprover_methods_to_ignore.count(id2string(method.name))) ||
```

Comment:
nit-pick: Visual Studio would like to have an explicit `!= 0` with `.count(...)`.